### PR TITLE
Various fixes to broken optional patches

### LINKED ISSUE #11: Applying "Original HUD.ips" causes a softlock upon entering the cave (only with MMC5 version)
Update: I seem to have fixed the issue, please see the pull request.

Thanks for this very impressive hack. I believe I have encountered a bug, though:

Emulators: FCEUX 2.6.1 and Mesen 0.9.9
Steps to reproduce:
Use base ROM: "Legend of Zelda, The (USA).nes" (CRC-32: D7AE93DF, MD5: 337bd6f1a1163df31bf2633665589ab0)
Apply "Zelda1_Redux_MMC5.ips" (CRC-32 of IPS: 09326F7D)
Apply "Original HUD.ips" (CRC-32 of IPS: 0F6FFF2A) from the MMC5 version's optional patches folder.
(Final Bugged ROM should have CRC-32: DD59ACC5)

// The game seems to play fine until the cave is entered. 
// It then softlocks with some wild graphical glitching, as the images below show:

FCEUX 2.6.1:
Legend of Zelda, The (USA)-1

Mesen 0.9.9:
Legend of Zelda, The (USA)_000

// This softlock also seems to corrupt the SRAM, as the game stays stuck on the grey loading screen on FCEUX reload.
// On Mesen, reloading the game causes this:

Legend of Zelda, The (USA)_002

// (None of this appears to be an issue with the MMC1 version.)

### LINKED ISSUE #12: Applying "Original NES GFX.ips" causes the game to not start, "Link's Awakening GFX.ips" graphical glitches (both only with MMC1 version)
Update3: Branched the repository and submitted a pull request.

UPDATE2: Found the fix.
In src/MMC1/code/optional/OriginalGFX.asm, change line 36 from:

`org $AE9B	// 0x0EEAB`
to
`org $AEA0	// 0x0EEB0`

The location of the animated tile frames had been shifted a few bytes between 2.0 and 3.0.

----------------------

Emulators: FCEUX 2.6.1 and Mesen 0.9.9
Steps to reproduce:
Use base ROM: "Legend of Zelda, The (USA).nes" (CRC-32: D7AE93DF, MD5: 337bd6f1a1163df31bf2633665589ab0)
Apply "Zelda1_Redux.ips" (CRC-32 of IPS: EFAFF81D)
Apply "Original NES GFX.ips" (CRC-32 of IPS: A8A89591) from the MMC1 version's optional patches folder.
(Final bugged ROM should have CRC-32: 71752410)

// The game will not start when loaded in either emulator.
// (This does not appear to be an issue with the MMC5 version.)

// update: compiling from current source with 
`incsrc code/optional/OriginalGFX.asm`
// active in optional.asm does not fix the problem.

### PR DESCRIPTION
**Pull request by auburnbit**
Please see commits for details.
(fixes #11, fixes #12)

### CODE REVIEW

**Comment by ShadowOne333** [comment]
Sorry for the very late reply, I did see your issue reports but haven't been able to respond to them yet due to vacations.
So far it all seems good, the only change I have doubts about is the one inside the Original HUD code for MMC5 ([MMC5/code/optional/OriginalHUD.asm](https://github.com/ShadowOne333/The-Legend-of-Zelda-Redux/pull/13/files#diff-f329137fc9aec40cf79eb77adae43da52e3fea2e57c45c20a83e1ae0997a3fee))

Have you verified and confirmed that commenting out those PPU related opcodes doesn't have any undesired drawbacks in-game?
Specifically these ones:
Original code from Move maps (MMC5):
https://github.com/ShadowOne333/The-Legend-of-Zelda-Redux/blob/master/MMC5/code/gameplay/move_maps.asm#L36-L49

Your change:
```
//org $9DC5	// 0x19DD5
//	adc.b #$62	// Starting PPU low byte
//org $9E22	// 0x19E32
//	ldy.b #$62	// Starting PPU low byte
//org $9E3D	// 0x19E4D
//	cmp.b #$F6	// Finishing PPU low byte
```

The one change at 9E3D doesn't seem to matter (since it's the same in both instances), but the other 2 have me in doubt as I believe these ones are the ones that set the starting X position for the Automap for MMC5. The original game used 2062 in PPU, while Bogaa's MMC5 implementation of Automap used 2076.

Once this one is confirmed, the rest can be easily be merged.
Thanks for taking a look at this!

**Comment by auburnbit** [comment]
Agreed on the uncertainty on the Original HUD code fix. I can say that it does stop the crash and allows the game to play, so I guess it's better than the alternative. However, I haven't done extensive testing or paid close attention to whether everything about the Automap position is correct. I'll play around with it some more and see what I can figure out.

**Comment by auburnbit** [comment]
It looks like the one change at 9E3D was the problem, commenting that out alone seems to stop the crash. I'm curious why modifying 9E3D makes it crash when Move Maps seems to make the same change without issue.

**Comment by auburnbit** [comment]
Wait... I bet $9E3D's supposed to be 
`cmp.b #$E2`
It's working as it is with that line commented out but I'll test it with the above change instead.

Edit: Yep that was it. I've updated the commit with the details. 

Edit2: Also double checked the three new IPS files to be sure I didn't mess something up and they seem to be correct. Also, it appears that there actually is no need to comment move_maps.asm out of main.asm when using OriginalHUD.asm (as it is now at least), as it overwrites everything move_maps.asm does (in both MMC1 and MMC5). As long as that remains true in future iterations of both files, there shouldn't be a need to comment out move_maps.asm when using Original.HUD.asm. I've removed the warnings I had added to the optional.asm files and updated the commit.

(I apologize for all the edits and force pushes, this is my first time actually using GitHub so I'm still figuring things out.)

File: MMC5/code/optional/OriginalHUD.asm [hunk 1]
```diff
@@ -86,7 +86,7 @@ org $9DC5	// 0x19DD5
 org $9E22	// 0x19E32
 	ldy.b #$62	// Starting PPU low byte
 org $9E3D	// 0x19E4D
-	cmp.b #$F6	// Finishing PPU low byte
```
Comment (auburnbit) [inline MMC5/code/optional/OriginalHUD.asm:89]: #$F6 comes from adding #$80 to #$76 (the starting ppu low byte in move_maps.asm) and wasn't updated when the starting ppu low byte was changed to #$62 in this file, causing a crash when entering the cave

File: MMC5/code/optional/OriginalHUD.asm [hunk 2]
```diff
@@ -86,7 +86,7 @@ org $9DC5	// 0x19DD5
 org $9E22	// 0x19E32
 	ldy.b #$62	// Starting PPU low byte
 org $9E3D	// 0x19E4D
-	cmp.b #$F6	// Finishing PPU low byte
+	cmp.b #$E2	// Finishing PPU low byte
```
Comment (auburnbit) [inline MMC5/code/optional/OriginalHUD.asm:89]: adding #$80 to #$62 gives us the correct finishing PPU low byte, #$E2, and fixes the crash

**Comment by ShadowOne333** [comment]
Gave these a look with the new modifications, and it all seems good!
Thank you so much for narrowing down the issues and debugging them to find a proper fix for these.
I'll merge this and update the optional patches in Romhacking.net as well with your fixes, I'll also add you in the credits for the fixes.

Just to be sure, the only IPS files modified with Link's Awakening GFX, Original GFX (MMC1) and Original HUD (MMC5), right?

**Comment by auburnbit** [comment]
Yes, that should be the only IPS files modified. Thanks!

**Pull request merged**